autoload/lsp/ui/vim/highlights.vim

```diff
@@ -95,11 +95,11 @@ function! s:place_highlights(server_name, path, diagnostics) abort
 
     if !empty(a:diagnostics) && l:bufnr >= 0
         for l:item in a:diagnostics
-            let l:line = l:item['range']['start']['line'] + 1
+            let l:line = l:item['range']['start']['line']
```

Comment:
```suggestion
            let l:line = l:item['range']['start']['line'] + 1
```

Reply:
I can't suggest the change from here, but instead you need to pass `l:line-1` to `nvim_buf_add_highlight` in line 106.